src/dominion/player.rs

```diff
@@ -54,14 +74,213 @@ impl Player {
             self.shuffle_deck();
         }
 
+        // TODO handle empty deck
         self.hand.push(self.deck_pile.remove(0));
     }
 
     pub fn cleanup(&mut self) {
+        self.discard_pile.append(&mut self.in_play);
         self.discard_pile.append(&mut self.hand);
 
         for _ in 0..5 {
             self.draw_card();
         }
     }
+
+    pub fn start_turn(&mut self) -> Result<()> {
+        if let None = self.phase {
```

Comment:
Well, I don't know rust :) but this (`if let None = ...`) strikes me as very odd. I'm assuming setting `self.phase` to `Action`. But why do it like this?

Reply:
The idea is that a player should only be able to start a turn if that player's turn is not already active.

```rust
if let None = self.phase {
    // do stuff
}
```
So here we check if `self.phase` is `None` and if it is, then do stuff. If it is `Some`, we don't really care what is contained in the `self.phase` option and simply return an error.

Reply:
Of course, I get it now. Thanks!

Reply:
There is a method on Option to do exactly this called `is_none`.